python-checks/src/main/java/org/sonar/python/checks/PandasAddMergeParametersCheck.java

```diff
@@ -50,24 +88,39 @@ private static void verifyMergeCallParameters(SubscriptionContext ctx) {
     CallExpression callExpression = (CallExpression) ctx.syntaxNode();
     Optional.ofNullable(callExpression.calleeSymbol())
       .map(Symbol::fullyQualifiedName)
-      .filter(MERGE_METHODS::contains)
-      .ifPresent(fqn -> missingArguments(ctx, callExpression));
+      .filter(METHODS::contains)
+      .ifPresent(fqn -> missingArguments(fqn, ctx, callExpression));
   }
 
-  private static void missingArguments(SubscriptionContext ctx, CallExpression callExpression) {
+  private static void missingArguments(String fullyQualifiedName, SubscriptionContext ctx, CallExpression callExpression) {
     List<String> parameters = new ArrayList<>();
-    if (TreeUtils.argumentByKeyword("how", callExpression.arguments()) == null) {
-      parameters.add("how");
+    if (argumentIsMissing(fullyQualifiedName, Keywords.HOW, callExpression.arguments())) {
+      parameters.add(Keywords.HOW.getKeyword());
     }
-    if (TreeUtils.argumentByKeyword("on", callExpression.arguments()) == null) {
-      parameters.add("on");
+    if (argumentIsMissing(fullyQualifiedName, Keywords.ON, callExpression.arguments())) {
+      parameters.add(Keywords.ON.getKeyword());
     }
-    if (TreeUtils.argumentByKeyword("validate", callExpression.arguments()) == null) {
-      parameters.add("validate");
+    if (argumentIsMissing(fullyQualifiedName, Keywords.VALIDATE, callExpression.arguments())) {
+      parameters.add(Keywords.VALIDATE.getKeyword());
     }
     if (!parameters.isEmpty()) {
-      ctx.addIssue(callExpression, String.format(messages.get(parameters.size() - 1), parameters.toArray()));
+      parameters.add(fullyQualifiedName.substring(fullyQualifiedName.lastIndexOf('.') + 1));
+      ctx.addIssue(callExpression, String.format(MESSAGES.get(parameters.size() - 2), parameters.toArray()));
```

Comment:
As an additional nitpick, I'm not a fan of the arithmetics to get the list index in `MESSAGES.get(parameters.size() - 2)`.
I'd probably prefer an explicit computation of the number of missing arguments and having the message retrieved from something like a map. I admit it sounds a bit overkill for such a simple case, but I think it would lead to more maintainable code.

Reply:
I changed the way that it was implemented in 57fbeb2d6285498a551e7fbf581e7c47ab35da99.